src/Symfony/Component/Process/ExecutableFinder.php

```diff
@@ -48,9 +48,14 @@ public function addSuffix(string $suffix)
      */
     public function find(string $name, ?string $default = null, array $extraDirs = [])
     {
+        $dirs = array_merge(
+            explode(\PATH_SEPARATOR, getenv('PATH') ?: getenv('Path')),
+            $extraDirs
+        );
+
         if (\ini_get('open_basedir')) {
             $searchPath = array_merge(explode(\PATH_SEPARATOR, \ini_get('open_basedir')), $extraDirs);
-            $dirs = [];
+
```

Comment:
With the new logic `$extraDirs` do not have to be searched here. Imho https://github.com/symfony/symfony/pull/57954 is the correct variant.

Reply:
#57954 which keeps adding the open_basedir folders as dirs to search in does not make sense.

Reply:
@stof so does this PR now. Symfony 6 and 7 also do not check whether the paths are allowed by `open_basedir` - the check happens by silencing `open_basedir` errors when using `is_dir` and `is_executable` with `@`. See

* https://github.com/symfony/symfony/blob/a8098b242ba3889c4f3c5b648dcd3e355af51ba5/src/Symfony/Component/Process/ExecutableFinder.php#L72
* https://github.com/symfony/symfony/blob/a8098b242ba3889c4f3c5b648dcd3e355af51ba5/src/Symfony/Component/Process/ExecutableFinder.php#L76
* https://github.com/symfony/symfony/blob/a8098b242ba3889c4f3c5b648dcd3e355af51ba5/src/Symfony/Component/Process/ExecutableFinder.php#L89